src/Shared/WeakStringCache.Concurrent.cs

```diff
@@ -65,39 +69,37 @@ public WeakStringCache()
             // We don't have the string in the cache - create it.
             result = internable.ExpensiveConvertToString();
 
-            // If the handle is new, we have to add it to the cache. We do it after removing unused handles if our heuristic
-            // indicates that it would be productive. Note that the _capacity field accesses are not protected from races. Being
-            // atomic (as guaranteed by the 32-bit data type) is enough here.
+            // Set the handle to reference the new string and put it in the cache.
+            handle.SetString(result);
+            if (!_stringsByHashCode.TryAdd(hashCode, handle))
+            {
+                // If somebody beat us to it and the handle has not been added, free it.
```

Comment:
tiny nit:
has now been added?

Reply:
Thanks, yes, the sentence is confusing. I have changed it to
```C#
// Another thread has managed to add a handle for the same hash code, so the one we got can be freed.
```
same as a few lines above where the same pattern is repeated.